packages/statsd/src/client.ts

```diff
@@ -139,6 +139,23 @@ export class StatsDClient {
     });
   }
 
+  incrementByAmount(
```

Comment:
So I think you can do this:
```typescript
  increment(stat: string | string[], tags?: Tags, options?: MetricOptions, value?: number);
  increment(stat: string | string[], tags?: Tags, options: MetricOptions = {}, value: number = 1) {
    return new Promise<void>((resolve) => {
      this.statsd.increment(
        stat,
        value,
        options.sampleRate,
        this.normalizeTags(tags),
        this.createCallback(resolve),
      );
    });
  }
```
To overload the method we have and not break the existing client usage.

Reply:
I added [35bd84e](https://github.com/Shopify/quilt/pull/2566/commits/35bd84ebdc7525f1a1c90b3c4469abc21822dc84) with the changes. I think there's pros and cons to both, but the former (`incrementByAmount`) is a bit more explicit.

I personally prefer `increment` but just can't seem to get it clean enough, the current setup will require an empty metric options array if you choose to add a value in, i.e.

```ts
increment('metric', ['tags'], {}, 5)
```

Let me know what you think!

Reply:
I wish we'd used the approach of named parameters with a destructured object for these API for this very reason. It's much easier to evolve an API that way.